### PR TITLE
chore: add example for a simple regex match on an input

### PR DESCRIPTION
**Pull request by lukeocodes**
I hope you don't mind. I thought examples/ could do with a few more basic prompts in there. This is the first of a few I was thinking of. I will PR the others when I get time.

### CODE REVIEW

File: packages/inquirer/examples/regex-validate-input.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+/* eslint-disable no-promise-executor-return */
+const questions = [
+  {
+    type: 'input',
+    name: 'api_key',
+    message: 'Please enter a valid API key.',
+    validate(input) {
+      return Promise.resolve().then(() => {
+        if (!!input.match(/([a-f0-9]{40})/g)) {
```
Comment (SBoudrias) [inline packages/inquirer/examples/regex-validate-input.js:16]:
```suggestion
        if (/([a-f0-9]{40})/g.test(input)) {
```

Getting the type we want directly instead

**Comment by SBoudrias** [comment]
Any reason to return a Promise in the example since the effective check is synchronous?

**Comment by lukeocodes** [comment]
> Any reason to return a Promise in the example since the effective check is synchronous?

It seemed to suggest it preferred a promise as a response! I will change it

**Comment by SBoudrias** [comment]
Yes and no, maybe examples are often async to be more easily extendable. Not against keeping the promise there, I was just wondering the rationale.

**Comment by lukeocodes** [comment]
I think it is easier to understand without the promise so I've switched it.

**Comment by lukeocodes** [comment]
Can you approve the workflow and i'll resolve the linting from the latest version. Thanks!

**Review by SBoudrias — APPROVED** [review]

**Comment by SBoudrias** [comment]
🎉 thanks!

**Pull request merged**